src/overlays/actors/ovl_En_Demo_heishi/z_en_demo_heishi.c

```diff
@@ -31,37 +39,150 @@ const ActorInit En_Demo_heishi_InitVars = {
     (ActorFunc)EnDemoheishi_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80BE9A50 = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 40, 40, 0, { 0, 0, 0 } },
 };
 
-#endif
-
-extern ColliderCylinderInit D_80BE9A50;
-
-extern UNK_TYPE D_06003BFC;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/EnDemoheishi_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/EnDemoheishi_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE95EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE9678.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE970C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE975C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE97F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE980C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/EnDemoheishi_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE9974.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/EnDemoheishi_Draw.s")
+static u16 sTextIds[] = { 0x1473 };
+
+void EnDemoheishi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDemoheishi* this = THIS;
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 25.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600D640, &D_06003BFC, this->jointTable, this->morphTable, 17);
+    this->actor.colChkInfo.mass = 0xFF;
+    this->actor.targetMode = 6;
+    this->actor.gravity = -3.0f;
+    Collider_InitAndSetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+    EnDemoheishi_SetupIdle(this);
+}
+
+void EnDemoheishi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDemoheishi* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+}
+
+void EnDemoheishi_ChangeAnimation(EnDemoheishi* this, s32 animIndex) {
+    static AnimationHeader* sAnimations[] = {
+        (AnimationHeader*)0x06006C18, (AnimationHeader*)0x06002A84, (AnimationHeader*)0x06003BFC,
+        (AnimationHeader*)0x06003380, (AnimationHeader*)0x06004770,
```

Comment:
Since we know what object this actor uses in the ActorInitVars up above, we can make these references instead of hard coded pointers.

The object file is found `<MM>/assets/xml/objects/object_sdn.xml` which contains all of the animations in this list, since this is your first PR you probably don't have to rename them with documentation, but you should still link to them, and then at the top you have to add `#include "objects/object_ani/object_sdn.h"`

Reply:
This was interesting. I went through past PRs and read through the object decomp tutorials for OOT decomp and learned some stuff. I pushed changes where I documented the object xml file, ran ./extract-assets and linked the header and objects in the actor so that we're no longer using hardcoded pointers. Recently pushed my changes, did I handle that right?